upgrade/upgrade-provider.go

```diff
@@ -27,7 +28,15 @@ func UpgradeProvider(ctx Context, name string) error {
 	var targetBridgeVersion string
 	var upgradeTargets UpstreamVersions
 	var goMod *GoMod
-	upstreamProviderName := strings.TrimPrefix(name, "pulumi-")
+
+	tok := strings.Split(name, "/")
+	if len(tok) != 2 {
+		return errors.New("argument must be provided as {org}/{repo}")
+	}
```

Comment:
Can we move this validation to the part of the repo that handles CLI interaction. We can then change `UpgradeProvider` to take `(ctx Context, repoOrg, repoName string)`.